xla/pjrt/pjrt_stream_executor_client.h

```diff
@@ -850,6 +924,17 @@ class PjRtStreamExecutorLoadedExecutable : public PjRtLoadedExecutable {
     if (proto != nullptr) {
       memory_stats.serialized_hlo_proto = proto->SerializeAsString();
     }
+    // TODO: Should this code live in backend-specific subclasses?
```

Comment:
This code used to be used as part of multiple in-tree backends, but at this point the only in-tree backend is GPUs (NVIDIA and AMD). There's actually a question here I can't answer: is anyone using this code for an out-of-tree backend, other than GPU? It's conceivable that someone is using this as a basis of an out-of-tree PJRT plugin.

I might merge this with the `dynamic_cast` but leave a TODO to specialize this code once this question is resolved.

Reply:
👍 Added a comment to that effect.

Do I understand correctly that you're suggesting to remove the `dynamic_cast<xla::cpu::CpuExecutable*>` entirely?